pallets/eth-bridge/src/tests/tests.rs

```diff
@@ -90,3 +96,506 @@ fn run_checks(
+            setup_active_transaction_data(Some(|active_tx| {
+                active_tx.data.eth_tx_hash = H256::repeat_byte(1);
+            }));
+    
+            let result = EthBridge::add_eth_tx_hash(
+                RawOrigin::None.into(),
+                tx_id,
+                context.eth_tx_hash,
+                context.author,
+                context.test_signature.clone(),
+            );
+    
+            assert_eq!(result, Err(Error::<TestRuntime>::EthTxHashAlreadySet.into()));
+        });
+    }
+    
+    #[test]
+    fn eth_tx_hash_set_by_sender_error() {
+        let mut ext = ExtBuilder::build_default().with_validators().as_externality();
+        ext.execute_with(|| {
+            let context = setup_context();
+            let tx_id = setup_eth_tx_request(&context);
+    
+            setup_active_transaction_data(Some(|active_tx| {
+                active_tx.data.sender = Default::default();
+            }));
+    
+            let result = EthBridge::add_eth_tx_hash(
+                RawOrigin::None.into(),
+                tx_id,
+                context.eth_tx_hash,
+                context.author,
+                context.test_signature.clone(),
+            );
+    
+            assert_eq!(result, Err(Error::<TestRuntime>::EthTxHashMustBeSetBySender.into()));
+        });
+    }
+    #[test]
+    fn eth_tx_hash_set_correctly() {
+        let mut ext = ExtBuilder::build_default().with_validators().as_externality();
+        ext.execute_with(|| {
+            let context = setup_context();
+            let tx_id = setup_eth_tx_request(&context);
+    
+            let result = EthBridge::add_eth_tx_hash(
+                RawOrigin::None.into(),
+                tx_id,
+                context.eth_tx_hash,
+                context.author,
+                context.test_signature.clone(),
+            );
+    
+            assert_eq!(result, Ok(().into()));
+        });
+    }
+}
+
+#[cfg(test)]
+mod add_corroboration {
+    use super::*;
+    use frame_support::assert_ok;
+    use frame_system::RawOrigin;
+    fn setup_corroboration_test(
+        context: &Context,
+        is_tx_successful: bool,
+        is_hash_valid: bool,
+    ) -> ActiveTransactionData<TestRuntime> {
+        let tx_id = setup_eth_tx_request(&context);
+
+        assert_ok!(EthBridge::add_corroboration(
+            RawOrigin::None.into(),
+            tx_id,
+            is_tx_successful,
+            is_hash_valid,
+            context.confirming_author.clone(),
+            context.test_signature.clone(),
+        ));
+
+        ActiveTransaction::<TestRuntime>::get().expect("Active transaction should be present")
+    }
+
+    #[test]
+    fn adds_invalid_hash_and_successful_corroboration_correctly() {
+        let mut ext = ExtBuilder::build_default().with_validators().as_externality();
+        ext.execute_with(|| {
+            let context = setup_context();
+            let active_tx = setup_corroboration_test(&context, true, true);
```

Comment:
I would inline the contents of this function in the tests and correctly identify the parameters as `valid_tx_hash` and `invalid_tx_hash` (same for the result) instead of passing in booleans